modules/generic-scala3/src/main/scala/pureconfig/generic/scala3/AnyValDerivationMacros.scala

```diff
@@ -0,0 +1,78 @@
+package pureconfig
+package generic
+package scala3
+
+import scala.compiletime._
+import scala.deriving.Mirror
+import scala.quoted._
+
+private[scala3] object AnyValDerivationMacros {
+
+  /** Derive a `ConfigReader` for a value class. Can only be called after checking that `A` is a value class.
+    */
+  inline def unsafeDeriveAnyValReader[A](derivationImpl: HintsAwareConfigReaderDerivation): ConfigReader[A] =
```

Comment:
Passing `derivationImpl` here enables the usage of the implementation provided by the `HintsAwareConfigReaderDerivation` trait, which hypothetically could be be further extended with some custom logic, for example - default static hints instead of summoned inline ones.

Reply:
I removed it, it's indeed not yet used internally. 

The main use case for that was going to be implementing an internal variant of derivation without summoning inline hints that could unblock the `derives` syntax for `pureconfig-generic-scala3`, but that was going to be in some subsequent PR.
When experimenting with it I noticed that the mechanism is quite generic and flexible which might be useful to users, but you are right - it's better to use hints for this.
However, I strongly believe that if something is not intended to be used by users directly then we ought to make it private, even if it's less convenient!

Reply:
Thanks for changing it!

> However, I strongly believe that if something is not intended to be used by users directly then we ought to make it private, even if it's less convenient!

I don't think functions and classes were deliberately left public, they're public only because no one bothered much with visibility modifiers. I'm aligned with you, feel free to make all derivation internals private - I'd personally not treat it as a breaking change.